src/Tests/dotnet-workload-install.Tests/WorkloadGarbageCollectionTests.cs

```diff
@@ -0,0 +1,393 @@
+
+            CreateManifestRecord("testmanifest", "1.0.0", "6.0.100", "6.0.300");
+            CreateManifestRecord("testmanifest", "2.0.0", "6.0.300", "6.0.300");
+            CreateManifestRecord("testmanifest", "3.0.0", "6.0.300", "6.0.300");
+
+            CreateInstallState("6.0.300",
+               """
+                {
+                    "manifests": {
+                        "testmanifest": "2.0.0/6.0.300",
+                    }
+                }
+                """);
+
+            var (installer, getResolver) = GetTestInstaller("6.0.300");
+
+            var packsToKeep = new PackInfo[]
+            {
+                CreatePackInfo("Xamarin.Android.Sdk", "8.4.7", WorkloadPackKind.Sdk),
+                CreatePackInfo("Xamarin.Android.Framework", "8.5.0", WorkloadPackKind.Framework),
+                CreatePackInfo("Xamarin.Android.Runtime", "8.5.0.1", WorkloadPackKind.Library)
+            };
+
+            var packsToCollect = new PackInfo[]
+            {
+                CreatePackInfo("Xamarin.Android.Framework", "8.6.0", WorkloadPackKind.Framework),
+                CreatePackInfo("Xamarin.Android.Runtime", "8.6.0.0", WorkloadPackKind.Library)
+            };
+
+            foreach (var pack in packsToKeep.Concat(packsToCollect))
+            {
+                CreateInstalledPack(pack, "6.0.300");
+            }
+
+            // Write workload install record for 6.0.300
+            var workloadsRecordPath = Path.Combine(_dotnetRoot, "metadata", "workloads", "6.0.300", "InstalledWorkloads");
+            Directory.CreateDirectory(workloadsRecordPath);
+            File.Create(Path.Combine(workloadsRecordPath, "xamarin-android-build"));
+
+            installer.GarbageCollect(getResolver);
+
+            foreach (var pack in packsToCollect)
+            {
+                PackShouldExist(pack, false);
+                PackRecord(pack, "6.0.300").Should().NotExist();
+            }
+            foreach (var pack in packsToKeep)
+            {
+                PackShouldExist(pack, true);
+                PackRecord(pack, "6.0.300").Should().Exist();
+            }
+
+            ManifestRecord("testmanifest", "1.0.0", "6.0.100", "6.0.300").Should().NotExist();
+            ManifestRecord("testmanifest", "2.0.0", "6.0.300", "6.0.300").Should().Exist();
+            ManifestRecord("testmanifest", "3.0.0", "6.0.300", "6.0.300").Should().NotExist();
+
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.100", "testmanifest", "1.0.0", "WorkloadManifest.json")).Should().NotExist();
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.300", "testmanifest", "2.0.0", "WorkloadManifest.json")).Should().Exist();
+            new FileInfo(Path.Combine(_dotnetRoot, "sdk-manifests", "6.0.300", "testmanifest", "3.0.0", "WorkloadManifest.json")).Should().NotExist();
+        }
+
+        //  Additional scenarios to add tests for once workload sets are added:
```

Comment:
Sorry for not being in the loop here, but the workload set garbage collection code has been merged, but is that not taking affect yet?  I don't see a work item to make these tests here, either: https://github.com/dotnet/sdk/issues/35455

Reply:
The garbage collection for side-by-side manifests has been merged.  Garbage collection for workload sets themselves is only partially implemented, and we don't yet create or install workload sets.